sdk/src/main/scala/io/horizen/account/state/ForgerStakeStorageV3.scala

```diff
@@ -0,0 +1,404 @@
+package io.horizen.account.state
+
+import com.google.common.primitives.Bytes
+import io.horizen.account.proposition.{AddressProposition, AddressPropositionSerializer}
+import io.horizen.account.state.ForgerStakeStorageV3.ACCOUNT
+import io.horizen.account.state.NativeSmartContractMsgProcessor.NULL_HEX_STRING_32
+import io.horizen.account.utils.BigIntegerUInt256
+import io.horizen.account.utils.WellKnownAddresses.FORGER_STAKE_V2_SMART_CONTRACT_ADDRESS
+import io.horizen.evm.Address
+import io.horizen.proposition.{PublicKey25519Proposition, VrfPublicKey}
+import io.horizen.utils.BytesUtils
+import sparkz.core.serialization.{BytesSerializable, SparkzSerializer}
+import sparkz.crypto.hash.Blake2b256
+import sparkz.util.serialization.{Reader, Writer}
+
+import java.math.BigInteger
+import scala.collection.mutable.ListBuffer
+
+
+object ForgerStakeStorageV3 {
+
+  val ACCOUNT = FORGER_STAKE_V2_SMART_CONTRACT_ADDRESS
+
+  def getForgerKey(blockSignProposition: PublicKey25519Proposition, vrfPublicKey: VrfPublicKey): Array[Byte] =
+    Blake2b256.hash(Bytes.concat(blockSignProposition.pubKeyBytes(), vrfPublicKey.pubKeyBytes()))
+
+  def getDelegatorKey(delegatorPublicKey: Address): Array[Byte] =
+    Blake2b256.hash(new AddressProposition(delegatorPublicKey).pubKeyBytes())
+
+  def getStakeKey(forgerKey: Array[Byte], delegatorPublicKey: Address): Array[Byte] = {
+    val delegatorKey = getDelegatorKey(delegatorPublicKey)
+    Blake2b256.hash(Bytes.concat(forgerKey, delegatorKey))
+  }
+
+  def addForger(view: BaseAccountStateView,
```

Comment:
please doulbe check (maybe with a unit test) in all the methods that the behaviour is not influenced by the case in the addresses/keys
(if I change the case of some characters in the address for example I should not end up with 2 different records)

Reply:
This is a very good point and a real issue, thanks for finding it! I have changed the code so the checksum address is always used for the delegator.